src/recommendations/recommender.py

```diff
@@ -26,17 +27,31 @@ def get_recipes_sections(user_id):
     recipes = _recommend_recipes(user_id, conn, get_recipes()) + _get_cold_start_recipes(user_id, conn)
     recipes.sort(key=get_rank)
     conn.close()
-
     return recipes
 
 
+def get_recipes_with_connection_by_is_saved(user_id, is_saved):
+    conn = connect()
```

Comment:
after you execute the query you should close the connection, like \
`conn.close()` \
same for the other function